test/integration/externalnode_test.go

```diff
@@ -119,58 +163,97 @@ var _ = Describe(fmt.Sprintf("%s,%s: ExternalNode", focusAws, focusAzure), func(
 		return err
 	}
 
-	validateResult := func(expectedResult string, kubectlCmd string, agented bool, en *antreav1alpha1.ExternalNode,
-		ee *antreav1alpha2.ExternalEntity) {
-		var actualResult string
+	removeEmptyStr := func(input []string) []string {
+		var output []string
+		for _, vm := range input {
+			if vm != "" {
+				output = append(output, vm)
+			}
+		}
+		return output
+	}
+
+	checkEqual := func(actual, expected []string) bool {
+		if len(actual) != len(expected) {
+			logf.Log.Info("the value is not correct", "actual", actual, "expected", expected)
+			return false
+		}
+		sort.Slice(actual, func(i, j int) bool {
+			return strings.Compare(actual[i], actual[j]) < 0
+		})
+		sort.Slice(expected, func(i, j int) bool {
+			return strings.Compare(expected[i], expected[j]) < 0
+		})
+		for i, v := range actual {
+			if v != expected[i] {
+				logf.Log.Info("the value is not correct", "actual", actual, "expected", expected)
+				return false
+			}
+		}
+		return true
+	}
+
+	validateResult := func(expectedResult []string, kubectlCmd string, agented bool, enList []antreav1alpha1.ExternalNode,
+		eeList []antreav1alpha2.ExternalEntity) {
+		var actualResult []string
+		// Keep a copy of expectedResult as the content of slice gets reordered in checkEqual() function.
+		copyExpected := make([]string, len(expectedResult))
+		copy(copyExpected, expectedResult)
 
 		output, err := kubeCtl.Cmd(kubectlCmd)
 		Expect(err).ToNot(HaveOccurred())
-
 		// Trim ' from output and remove trailing "\n".
 		stringList := strings.Split(strings.Trim(output, "'"), "\n")
 		if len(stringList) > 0 {
-			actualResult = stringList[0]
+			actualResult = removeEmptyStr(stringList)
 		} else {
-			actualResult = ""
+			actualResult = nil
 		}
+
 		if agented == true {
-			logf.Log.Info("Validate External Node Name", "Actual", actualResult, "Expected", expectedResult)
-			Expect(actualResult == expectedResult).To(BeTrue())
-
-			// Validate spec field in the generated External Node CR.
-			spec := &antreav1alpha1.ExternalNodeSpec{
-				Interfaces: []antreav1alpha1.NetworkInterface{
-					{
-						Name: strings.ToLower(nic),
-						IPs:  []string{privateIp, publicIp},
+			logf.Log.Info("Validate generated External Node Name(s) with Expected",
```

Comment:
Keep less logging. This gets printed on console. Only for important steps and move other to debug

Reply:
Moved a few to debug logging